transcom/mymove
# MB-16469: Add shared definitions

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16469)

## Summary

This adds shared definitions for some of the definitions in the prime.yaml file. If you take a look at the generated files you can see that there are generally no actual changes (outside the readOnly attribute that was added). 

Adding these shared definitions would make it so that we better identify the objects that have been modified in a v2 version.

## Verification Steps for Reviewers

Ensure that the generated swagger files are effectively the same as before and that there is no functional changes to any of the APIs or generated code.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

There are a handful of ways to test this code. You can likely manually review the code and verify that there is no net changes. You can also compare the previous versions of the generated code by sorting all of the lines alphabetically and diffing the previous version with the new version. (You can do this either with your text editor or through command line tools.)

Eg. If you are trying to diff the files using vsCode you can do the following:
1. Save the contents of the previous generated swagger file say `swagger/prime.yaml` into a new file called `before.txt`. 
    - Then sort the contents of the file using the `Sort Lines Descending` command in vsCode.
2. Save the contents of the newly generated swagger file say `swagger/prime.yaml` into a new file called `after.txt`.
    - Then sort the contents of the file using the `Sort Lines Descending` command in vsCode.
3. Open the terminal and run `diff <path/to/before.txt> <path/to/after.txt>` where the contents within the <> are replaced
